tests/core/test_otel_logger.py

```diff
@@ -173,13 +171,54 @@ def test_decr_with_rate_limit_works(self, mock_random, name):
         self.map[full_name(name)].add.assert_has_calls(expected_calls)
         self.map[full_name(name)].add.call_count == 2
 
-    @mock.patch("warnings.warn")
-    def test_timer_warns_not_implemented(self, mock_warn):
-        with self.stats.timer():
-            mock_warn.assert_called_once_with("OpenTelemetry Timers are not yet implemented.")
+    def test_gauge_new_metric(self, name):
```

Comment:
Should there be a test for getting the value of a gauge? All the tests below seem to be about setting/creating them.

Reply:
I checked it [here](https://github.com/apache/airflow/pull/31725/files#diff-1a02f43043ead15894e9b58e32b1ed9a4a079ed461ab73b66f28d984baeab261R199) in the "test_gauge_existing_metric" test, but I can certainly add that check in to the "test_gauge_new_metric" test as well.  Never hurts to have more confirmation :+1:

Reply:
And I'll add a new test to verify that `delta` is working as intended :+1:    I think that should cover our bases on that?